lib/SPIRV/SPIRVLowerBitCastToNonStandardType.cpp

```diff
@@ -150,6 +150,13 @@ class SPIRVLowerBitCastToNonStandardTypePass
     // parameter, since it added by an optimization.
     bool Changed = false;
 
+    // SPV_INTEL_vector_compute allows to use vectors with any number of
+    // components. Since this method only lowers vectors with non-standard
+    // in pure SPIR-V number of components, there is no need to do anything in
+    // case SPV_INTEL_vector_compute is enabled.
+    if (Opts.isAllowedToUseExtension(ExtensionID::SPV_INTEL_vector_compute))
+      return PreservedAnalyses::all();
```

Comment:
It is out of scope of this patch, but it is strange that we return `PreservedAnalyses`, because it is not an analysis pass AFAIK,

Reply:
I was following the logic that was made here before. I'm a newbie in passes that run through new passmanager. Probably @KornevNikita can help to understand it better once returns from vacation.

Reply:
I suppose current method better to return just bool "changed or not".
In main run method PreservedAnalyses should be created depending on changed value.